docs/docs/monitoring/tracing.mdx

```diff
@@ -162,3 +171,26 @@ Observable `TrackerStore` and `LockStore` attributes include:
 - `number_of_streamed_events`: number of new events to stream
 - `broker_class`: the `EventBroker` on which the new events are published
 - `lock_store_class`: Name of lock store used to lock conversations while messages are actively processed
+
+## Tracing in the Action Server
+
+Requests are now being traced as they flow through the action server by
+instrumenting the webhook that receives custom actions.
+
+### Action server Attributes
+The following attributes are captured as part of the trace context;
+- `http.method`: the http method used to make the request
+- `http.route`: the endpoint of the request
+- `next_action`: the name of the next action to be executed
+- `version`: the rasa version used
+- `sender_id`: the conversation id
+- `message_id`: the unique message id
+
+You can also continue tracing the request further along
+your custom action code by [creating spans](https://opentelemetry.io/docs/instrumentation/python/manual/#creating-spans)
```

Comment:
Can you add an example of how can `span` be accessed inside the custom action so that `span.set_attribute('cuisine', cuisine)` or `with tracer.start_as_current_span(span_name, context=context) as span:` can also be done by anyone who is writing functions or custom actions with Rasa SDK?

```python
from typing import Text, Dict, Any, List
from rasa_sdk import Action
from rasa_sdk.events import SlotSet

class ActionCheckRestaurants(Action):
   def name(self) -> Text:
      return "action_check_restaurants"

   def run(self,
           dispatcher: CollectingDispatcher,
           tracker: Tracker,
           domain: Dict[Text, Any]) -> List[Dict[Text, Any]]:

      cuisine = tracker.get_slot('cuisine')
      span.set_attribute('cuisine', cuisine)
      q = "select * from restaurants where cuisine='{0}' limit 1".format(cuisine)
      result = db.query(q)

      return [SlotSet("matches", result if result is not None else [])]
```

Reply:
We can only add this when we instrument some classes in the action server. e.g (ValidationAction, FormValidationAction etc).
This will come in other iterations if requested for.